presto-main/src/main/java/io/prestosql/util/DateTimeUtils.java

```diff
@@ -143,58 +143,37 @@ public static String printDate(int days)
     }
 
     /**
-     * Parse a string (optionally containing a zone) as a value of either TIMESTAMP or TIMESTAMP WITH TIME ZONE type.
-     * <p>
-     * For example: {@code "2000-01-01 01:23:00"} is parsed to TIMESTAMP {@code 2000-01-01T01:23:00}
-     * and {@code "2000-01-01 01:23:00 +01:23"} is parsed to TIMESTAMP WITH TIME ZONE
-     * {@code 2000-01-01T01:23:00.000+01:23}.
-     *
-     * @return stack representation of TIMESTAMP or TIMESTAMP WITH TIME ZONE type, depending on input
+     * Parse a legacy TIMESTAMP type. The string is interpreted in {@code timeZoneKey} zone.
      */
-    public static long parseTimestampLiteral(String value)
+    @Deprecated
+    public static long parseLegacyTimestamp(TimeZoneKey timeZoneKey, String value)
     {
-        try {
-            DateTime dateTime = TIMESTAMP_WITH_TIME_ZONE_FORMATTER.parseDateTime(value);
-            return packDateTimeWithZone(dateTime);
-        }
-        catch (Exception e) {
-            return TIMESTAMP_WITHOUT_TIME_ZONE_FORMATTER.parseMillis(value);
-        }
+        return LEGACY_TIMESTAMP_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).parseMillis(value);
     }
 
     /**
-     * Parse a string (optionally containing a zone) as a value of either TIMESTAMP or TIMESTAMP WITH TIME ZONE type.
-     * If the string doesn't specify a zone, it is interpreted in {@code timeZoneKey} zone.
-     *
-     * @return stack representation of legacy TIMESTAMP or TIMESTAMP WITH TIME ZONE type, depending on input
-     * @deprecated applicable in legacy timestamp semantics only
+     * Parse a string (optionally containing a zone) as a value of TIMESTAMP type using legacy semantics.
```

Comment:
Specify in the javadoc what happens when

- timestamp is a gap in the provided zone (rejected or shifted)
- value contains a zone specifier -- is it converted to provided zone with "same zone" or "same instant" semantics
  - and what happens if it cannot be